### ISSUE TITLE
Running any of the scripts fails in event.js:85

### ISSUE THREAD

**Issue by cwillwer** [godaddy/javascript#16]
I have attempted to follow the directions for adding it to our project, but running any of the commands with `npm run <script>` results in

```
events.js:85
      throw er; // Unhandled 'error' event
            ^
```

I assume that events.js is from one of the node_modules that this depends on because we do not have an events.js file in our code. I changed the `lib/` to `src/` in the script command because we don't have a lib folder, and got the same behavior. npm error information looks like this:

```
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "c:\\Program Files\\nodejs\\node.exe" "c:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "lint"
npm ERR! node v0.12.7
npm ERR! npm  v2.11.3
npm ERR! code ELIFECYCLE
npm ERR! <our package name>@1.0.0 lint: `godaddy-js-style-lint src/ test/`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the <our package name>@1.0.0 lint script 'godaddy-js-style-lint src/ test/'.
npm ERR! This is most likely a problem with the <our package name> package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     godaddy-js-style-lint src/ test/
```

**Comment by gzurbach** [comment]
I think the `lib/` was just an example, you have to adjust it to your own use case.

There should be an error above, a call stack maybe. Could you post it?

**Comment by cwillwer** [comment]
Yes, that's why I said that I replaced `lib/` with `src/` because we have a `src` and `test` subdirectory parallel with our package.json file.

The callstack is:
```
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: spawn jscs ENOENT
    at exports._errnoException (util.js:746:11)
    at Process.ChildProcess._handle.onexit (child_process.js:1053:32)
    at child_process.js:1144:20
    at process._tickCallback (node.js:355:11)
    at Function.Module.runMain (module.js:503:11)
    at startup (node.js:129:16)
    at node.js:814:3
```

BTW, for people who are not knowledgeable in running npm scripts, it would be helpful if the documentation mentioned the `npm run <cmd>` command to run the script entries in the package.json file.

**Comment by gzurbach** [comment]
I've actually never used the `godaddy-js-style-lint`. Have you tried with `godaddy-js-style-eslint`?

**Comment by gshively11** [comment]
Hi @cwillwer, I think this is a windows-only bug that's manifesting in the [fashion-show](https://github.com/indexzero/fashion-show) package leveraged by godaddy-style (unconfirmed, but I've seen this behavior before).  The stack trace isn't conclusive, but we don't call spawn anywhere in godaddy-style.  I believe [this line](https://github.com/indexzero/fashion-show/blob/master/lib/run.js#L47) is the culprit.

When you have time, could you please create a system environment variable called `DEBUG` and set it's value equal to `*`, then rerun the npm script?  This should give you a bunch of debug output (you may need to restart your shell after creating the environment variable).  Please post a snippet of that output here.

**Comment by cwillwer** [comment]
Could very well be. Here's the sterilized debug output:
```
fashion-show:defaultify defaultify.defaults { commands: [ 'jscs', 'eslint' ],
  configDir: 'c:\\Development\\Repos\\...\\node_modules\\godaddy-style\\dist',
  targets: [ 'src/', 'test/' ],
  binPath: 'c:\\Development\\Repos\\...\\node_modules\\godaddy-style\\node_modules\\.bin',
  fix: true }
fashion-show:defaultify defaultify.options { commands: [ 'jscs', 'eslint' ],
  targets: [ 'src/', 'test/' ],
  rc: 'c:\\Development\\Repos\\...\\node_modules\\godaddy-style\\dist',
  cwd: 'c:\\Development\\Repos\\...',
  fix: true,
  env: {},
  binPath: 'c:\\Development\\Repos\\...\\node_modules\\godaddy-style\\node_modules\\.bin',
  reporter: undefined,
  format: undefined,
  global: undefined,
  exts: [] }
fashion-show env.$PATH c:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;c:\Development\Repos\...\node_modules\.bin;c:\Program Files\Java\jdk1.8.0_51\bin;c:\Use
rs\cwillwerth\bin;.;C:\Program Files (x86)\Git\local\bin;C:\Program Files (x86)\Git\mingw\bin;C:\Program Files (x86)\Git\bin;c:\Ruby22-x64\bin;c:\WINDOWS\system32;c:\WINDOWS;c:\WINDOWS\System32\Wbem;c:\WINDOWS\System32\WindowsPowerShell\v1.0\;c:\Program Files\Microsoft\Web Platform Installer\;c:\Developme
nt\tools;c:\WINDOWS\System32\WindowsPowerShell\v1.0\;c:\Development\tools\apache-maven-3.3.3\bin;c:\Program Files (x86)\HashiCorp\Vagrant\bin;c:\Program Files\nodejs\;c:\Program Files (x86)\Ruby22\bin;c:\Users\cwillwerth\AppData\Roaming\npm
fashion-show run.invoke { command: jscs }
fashion-show run.invoke { command: eslint }
fashion-show:argify config [ '-c',
  'c:\\Development\\Repos\\...\\node_modules\\godaddy-style\\dist\\.jscsrc' ]
fashion-show:argify lint-specific [ '--fix' ]
fashion-show:run:lint.bin jscs
fashion-show:run:lint.args ["-c","c:\\Development\\Repos\\...\\node_modules\\godaddy-style\\dist\\.jscsrc","--fix","src/","test/"]
fashion-show:run:lint.options {"cwd":"c:\\Development\\Repos\\..."}
Running jscs with options: -c c:\Development\Repos\...\node_modules\godaddy-style\dist\.jscsrc --fix src/ test/
fashion-show:argify config [ '-c',
  'c:\\Development\\Repos\\...\\node_modules\\godaddy-style\\dist\\.eslintrc' ]
fashion-show:argify lint-specific []
fashion-show:run:lint.bin eslint
fashion-show:run:lint.args ["-c","c:\\Development\\Repos\\...\\node_modules\\godaddy-style\\dist\\.eslintrc","src/","test/"]
fashion-show:run:lint.options {"cwd":"c:\\Development\\Repos\\..."}
Running eslint with options: -c c:\Development\Repos\...\node_modules\godaddy-style\dist\.eslintrc src/ test/
events.js:85
      throw er; // Unhandled 'error' event
...
```

**Comment by gshively11** [comment]
Just a quick update, I'm a bit swamped at the moment so I'll need a few more days before digging into this one.  Apologies for the delay.

**Comment by briboles** [comment]
This issue was resolved through PR in dependency `fashion-show` if anyone still sees this issue make sure you do a new `npm install` and you have `fashion-show` version `3.3.2` or higher.

https://github.com/indexzero/fashion-show/pull/10

**Comment by indexzero** [comment]
Fixed in #40.
